server/etcdserver/api/v3compactor/periodic.go

```diff
@@ -100,7 +100,7 @@ func (pc *Periodic) Run() {
 	compactInterval := pc.getCompactInterval()
 	retryInterval := pc.getRetryInterval()
 	retentions := pc.getRetentions()
-
+	prev := int64(0)
```

Comment:
This variable should be within the goroutine.

Reply:
done